File: Problem_Solving_Paradigms/Greedy_Approach/UVa_11292_The_Dragon_of_Loowater.py
# https://onlinejudge.org/external/112/11292.pdf

def solution(n, m, dragon, knight):
    dragon.sort()
    knight.sort()
    gold = d = k = 0
    while d < n and k < m:
        while dragon[d] > knight[k] and k < m:
            k += 1
        if k == m:
            break
        gold += knight[k]
        d += 1
        k += 1

    if d == n:
        return gold
    else:
        return "Loowater is doomed!"


while True:
    n, m = map(int, input().split())
    if not n and not m:
        break
    dragon = []
    knight = []
    for i in range(n):
        dragon.append(int(input()))
    for i in range(m):
        knight.append(int(input()))
    print(solution(n, m, dragon, knight))

File: UVa/Problem Solving/Greedy/UVa_11292_The Dragon of Loowater.cpp
#include <bits/stdc++.h>
using namespace std;
#define REP(i, n) for (int i = 0; i < n; i++)

int main()
{
    int n, m;
    int dragon[20000], knight[20000];
    int d, k, gold;

    while (scanf("%d %d", &n, &m), n || m)
    {
        REP(i, n)
        {
            scanf("%d", &dragon[i]);
        }
        REP(i, m)
        {
            scanf("%d", &knight[i]);
        }
        sort(dragon, dragon + n);
        sort(knight, knight + m);

        gold = d = k = 0;
        while (d < n and k < m)
        {
            while (k < m and dragon[d] > knight[k])
                ++k;
            if (k == m)
                break;
            gold += knight[k];
            ++d;
            ++k;
        }
        (d == n) ? printf("%d\n", gold) : printf("Loowater is doomed!\n");
    }
    return 0;
}